src/java/org/apache/cassandra/db/filter/ColumnFilter.java

```diff
@@ -182,67 +180,6 @@ boolean areAllFetchedColumnsQueried()
         abstract RegularAndStaticColumns getFetchedColumns(TableMetadata metadata, RegularAndStaticColumns queried);
     }
 
-    /**
-     * Returns {@code true} if there are pre-4.0-rc2 nodes in the cluster, {@code false} otherwise.
-     *
-     * <p>ColumnFilters from 4.0 releases before RC2 wrongly assumed that fetching all regular columns and not
-     * the static columns was enough. That was not the case for queries that needed to return rows for empty partitions.
-     * See CASSANDRA-16686 for more details.</p>
-     */
-    private static boolean isUpgradingFromVersionLowerThan40RC2()
-    {
-        if (Gossiper.instance.isUpgradingFromVersionLowerThan(CassandraVersion.CASSANDRA_4_0_RC2))
-        {
-            logger.trace("ColumnFilter conversion has been applied so that static columns will not be fetched because there are pre 4.0-rc2 nodes in the cluster");
```

Comment:
After getting rid of this code `ColumnFilter.logger` is not used anymore. We can get rid of it too.